### ISSUE TITLE
[ Bug ] The 2nd Saver fails to recognize its Checkpoint State file.

### ISSUE THREAD

**Issue by kumikokashii** [tensorflow/tensorflow#6596]
**Operating System:** macOS Sierra

**Steps to Reproduce:**
1. In a session, create two Savers.
2. Let one of the Saver save the variables.
3. Let the other Saver save the variables in a different directory from the first Saver's destination directory.

**Result:**
An Info message is displayed indicating that a Checkpoint State file does not exist (https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/training/saver.py#L738). And TensorFlow creates a new Checkpoint State file. This happens every time the 2nd Saver saves the variables.

**Expected Result:**
The 2nd Saver's Checkpoint State file should be recognized.

**Code to Reproduce:**
```
with tf.Session(graph=graph) as sess:
    ...
    saver_best = tf.train.Saver()
    saver_hourly = tf.train.Saver(max_to_keep=None)  
    ...
    for i in range(max_step):
        ...
        if last_hourly_save + datetime.timedelta(hours=1) < datetime.datetime.now():                     
            path_checkpoint_file = saver_hourly.save(sess, 'checkpoint_directory/hourly/model', global_step=i, latest_filename='hourly_checkpoint')
        ...
        if best:
            path_checkpoint_file = saver_best.save(sess, 'checkpoint_directory/best/model', global_step=i, latest_filename='best_checkpoint')
        ...
```

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by michaelisard** [comment]
Is this still an issue?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?
